corehq/apps/api/tests/test_case_update_api.py

```diff
@@ -0,0 +1,179 @@
+import uuid
+
+from django.test import TestCase
+
+from casexml.apps.case.mock import CaseBlock
+
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.users.models import WebUser
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors, FormAccessors
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+
+
+class TestUpdateCases(TestCase):
+    domain = 'test_update_cases'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(cls.domain)
+        cls.web_user = WebUser.create(cls.domain, 'netflix', 'password', None, None)
+        cls.case_accessor = CaseAccessors(cls.domain)
+        cls.form_accessor = FormAccessors(cls.domain)
+
+    def tearDown(self):
+        FormProcessorTestUtils.delete_all_cases(self.domain)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.domain_obj.delete()
+        super().tearDownClass()
+
+    def post(self, resource_url, body):
+        # TODO
+        pass
+
+    def test_create_case(self):
+        res = self.post('create_case', {
+            # notable exclusions: case_id, date_opened, date_modified
+            '@case_type': 'player',
+            '@case_name': 'Elizabeth Harmon',
+            '@owner_id': 'methuen_home',
+            'properties': {
+                'external_id': '1',
+                'sport': 'chess',
+                'dob': '1948-11-02',
+            },
+        })
+        self.assertEqual(res.keys(), ['@case_id', '@form_id'])
+        case = self.case_accessor.get_case(res['@case_id'])
+        self.assertEqual(case.domain, self.domain)
+        self.assertEqual(case.type, 'player')
+        self.assertEqual(case.name, 'Elizabeth Harmon')
+        self.assertEqual(case.owner_id, 'methuen_home')
+        self.assertEqual(case.opened_by, 'netflix')
+        self.assertEqual(case.external_id, '1')
+        self.assertEqual(case.dynamic_case_properties(), {
+            'dob': '1948-11-02',
+            'sport': 'chess',
+        })
+
+        xform = self.form_accessor.get_form(res['@form_id'])
+        self.assertEqual(xform.metadata.xmlns, 'http://commcarehq.org/case_api')
```

Comment:
Makes sense - I don't think there's a lot of utility in a totally generic xmlns, so making it in someway helpful would be nice.

Reply:
Would you then submit one form per case type? I think that could get complicated with parent/child relationships and the dependencies between them. 
@kaapstorm how would you be able to list case types with this? I think we would already be able to do that through ES (and that would be more accurate, since it would contain all cases, not just cases that are submitted through the API). That functionality does seem like scope creep for this though.

Reply:
Oh, man, I am sorry. My brain was on [the comment about API paths](https://github.com/dimagi/commcare-hq/pull/28884#discussion_r545315366), not XMLNS.

What's more, if the bulk update supports updating multiple case types, and we are updating up to 100 cases with the same XForm, then putting one case type in the XMLNS won't always be possible. We'd have to split the cases up by case type, and doing so for the sake of a nice XMLNS seems like the tail wagging the dog.

Reply:
(... which is exactly what you just said, @proteusvacuum ... I think I'll go do some Friday afternoon reading.)

Reply:
good point